aleo/detect.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) The Amphitheatre Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aleo
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+)
+
+const (
+	PlanEntryAleo = "aleo"
+)
+
+type Detect struct {
+}
+
+func (d Detect) Detect(context libcnb.DetectContext) (libcnb.DetectResult, error) {
+	found, err := d.aleoProject(context.Application.Path)
+	if err != nil {
+		return libcnb.DetectResult{}, fmt.Errorf("unable to detect Aleo requirements\n%w", err)
+	}
+
+	if !found {
+		return libcnb.DetectResult{Pass: false}, nil
+	}
+
+	return libcnb.DetectResult{
+		Pass: true,
+		Plans: []libcnb.BuildPlan{
+			{
+				Provides: []libcnb.BuildPlanProvide{
+					{Name: PlanEntryAleo},
+				},
+				Requires: []libcnb.BuildPlanRequire{
+					{Name: PlanEntryAleo},
+				},
+			},
+		},
+	}, nil
+}
+
+func (d Detect) aleoProject(appDir string) (bool, error) {
+	_, err := os.Stat(filepath.Join(appDir, "program.json"))
```

Comment:
Here you also have to check that the output of the previous compilation step is present